src/main/java/br/unicamp/cst/planning/PlanningCodelet.java

```diff
@@ -0,0 +1,62 @@
+package br.unicamp.cst.planning;
+
+import br.unicamp.cst.core.entities.Codelet;
+import br.unicamp.cst.core.entities.Memory;
+import br.unicamp.cst.core.exceptions.CodeletActivationBoundsException;
+
+public abstract class PlanningCodelet extends Codelet {
+
+    private String id;
+
+    private Memory inputInitialStateMO;
+    
+    private Memory inputGoalsMO;
+    private Memory inputActionsMO;
+    private Memory inputTransitionFunctionsMO;
+
+    private Memory outputPlanMO;
+
+    public PlanningCodelet(String id) {
+        setId(id);
+    }
+
+    @Override
+    public void accessMemoryObjects() {
+        if(inputInitialStateMO == null
+            && inputGoalsMO == null
+            && inputActionsMO == null
+            && inputTransitionFunctionsMO == null
+            && outputPlanMO == null
+        ){
+            inputInitialStateMO = getInput(PlanningMemoryNames.INPUT_INITIAL_STATE_MEMORY.toString());
+            inputGoalsMO = getInput(PlanningMemoryNames.INPUT_GOALS_MEMORY.toString());
+            inputActionsMO = getInput(PlanningMemoryNames.INPUT_ACTIONS_MEMORY.toString());
+            inputTransitionFunctionsMO = getInput(PlanningMemoryNames.INPUT_TRANSITION_FUNCTIONS_MEMORY.toString());
+            outputPlanMO = getOutput(PlanningMemoryNames.OUTPUT_PLAN_MEMORY.toString());
+        }
+    }
+
+    @Override
+    public void calculateActivation() {
+        try {
+            setActivation(0d);
+        } catch (CodeletActivationBoundsException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void proc() {
+        outputPlanMO.setI(planning(inputInitialStateMO, inputGoalsMO, inputActionsMO).getI());
```

Comment:
Looks good to me, but without a test that can be used also as a spec to this Codelet, it becomes quite hard to understand how all these abstract concepts should be used in practice. What do you think of adding a test to this PR, so anyone would have it as a spec for how to use this new PlanningCodelet?

Reply:
André I will implement tests using planning codelet in this PR, but I haven't had time to do that yet.